web/src/narrow.js

```diff
@@ -709,7 +768,9 @@ export function maybe_add_local_messages(opts) {
         // first unread message, or the target_id (if any), whichever
         // is earlier.  See #2091 for a detailed explanation of why we
         // need to look at unread here.
-        id_info.final_select_id = min_defined(id_info.target_id, unread_info.msg_id);
+        id_info.final_select_id = filter.has_operator("with")
+            ? unread_info.msg_id
+            : min_defined(id_info.target_id, unread_info.msg_id);
```

Comment:
Why is this the right special case?

Reply:
well, in case of this `with`  operator, we populate the `target_id` with the operand. The `final_select_id` in this case should always be the first unread, or if not found, then undefined (since we want the first unread to be the anchor).